### PR TITLE
Add first draft for a TypeScript support to the package

### PR DESCRIPTION
**Pull request by MartinHelmut**
## Description

This aims to add TypeScript support to mappersmith. Most of the action happens in `typings/index.d.ts` where the main types are defined. The `spec/typescript` folder is to run the `tsc` command against, to test the actual implemented types.

## ToDos

- [x] Local test in a TypeScript environment where mappersmith is used

### CODE REVIEW

**Comment by tulios** [comment]
Amazing PR @MartinHelmut ! I will start the review, thanks!

**Comment by MartinHelmut** [comment]
I tested the PR now against a local TypeScript project and I found one issue I will update now. Had to reference the sub-types inside the `index.d.ts` file.

File: README.md [hunk 1]
```diff
@@ -1147,6 +1148,64 @@ configs.gatewayConfigs.Fetch = {
...
+#### Create a middleware with TypeScript
+
+To create a middleware using TypeScript you just have to add the `Middleware` interface to your middleware object:
+
+```typescript
+import { Middleware } from 'mappersmith'
+
+const MyMiddleware: Middleware = () => ({
+  prepareRequest(next) {
+    return next().then(response => response.enhance({
```
Comment (tulios) [inline README.md]: `return next().then(request => request.enhance({`

small typo, it should be `request`.

Comment (MartinHelmut) [reply README.md]: Ohh yes, changed it!

**Comment by MartinHelmut** [comment]
Hey, is there a way to release a new version? :)

**Comment by tulios** [comment]
Yes, I will do it today. I can ping you once it's out

**Comment by tulios** [comment]
`2.29.0` is out

**Pull request merged**